### PR TITLE
in_memory_engine: adjust the default config

### LINKED ISSUE #17762: In-Memory-Engine's default configuration is not accurate

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #16141, close #17762

What's Changed:

```commit-message
Let in_memory_engine's config`evict-threshold` and `stop-load-threshold`
default value generated from `capacity`.
```

- in_memory_engine config default value:
  - `evict-threshold`: default value set to capacity - min(capacity * 0.1, 200MiB)
  - `stop-load-threshold`: default value set to: evict_threshold - min(capacity * 0.15, region_split_size * 2 + 200MiB)
  - set the minimal value of `load-evict-interval` to 120s as the background task will skip the check if the interval is small than 90s.
- Demote some logs in `LoadEvictObserver` from info to debug as RegionManager will also output these logs.
- Change 2 IME load/evcit metrics expression from `rate` to `delta` as the rate is too small and we care the actual event number than its rate.  

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/in_memory_engine/src/config.rs [hunk 1]
```diff
@@ -15,6 +15,16 @@ const MIN_GC_RUN_INTERVAL: Duration = Duration::from_secs(10);
 // The maximum interval for GC run is 10 minutes which equals to the minimum
 // value of TiDB GC lifetime.
 const MAX_GC_RUN_INTERVAL: Duration = Duration::from_secs(600);
+// the maximum write kv throughput(20MiB), this is an empiracal value.
```
Comment (overvenus) [inline components/in_memory_engine/src/config.rs:18]:
```suggestion
// the maximum write kv throughput(20MiB), this is an empirical value.
```

File: components/in_memory_engine/src/config.rs [hunk 2]
```diff
@@ -15,6 +15,16 @@ const MIN_GC_RUN_INTERVAL: Duration = Duration::from_secs(10);
...
+const MAX_WRITE_KV_SPEED: u64 = 20 * 1024 * 1024;
+// The maximun duration in seconds we expect IME to release enough memory after
```
Comment (overvenus) [inline components/in_memory_engine/src/config.rs:20]:
```suggestion
// The maximum duration in seconds we expect IME to release enough memory after
```

File: components/in_memory_engine/src/config.rs [hunk 3]
```diff
@@ -15,6 +15,16 @@ const MIN_GC_RUN_INTERVAL: Duration = Duration::from_secs(10);
...
+// memory usage reaches `evict_threshold`. This is an empiracal value.
+// We use this value to determine the default value of `evict_threshold` based
+// on `capacity`.
+const MAX_RESERVED_DURATION_FOR_WRITE: u64 = 10;
+// Regions' mvcc read amplificatoin statistics is updated every 1min, so we set
```
Comment (overvenus) [inline components/in_memory_engine/src/config.rs:25]:
```suggestion
// Regions' mvcc read amplification statistics is updated every 1min, so we set
```

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by SpadeA-Tang — APPROVED** [review]

File: components/in_memory_engine/src/config.rs [hunk 4]
```diff
@@ -15,6 +15,16 @@ const MIN_GC_RUN_INTERVAL: Duration = Duration::from_secs(10);
 // The maximum interval for GC run is 10 minutes which equals to the minimum
 // value of TiDB GC lifetime.
 const MAX_GC_RUN_INTERVAL: Duration = Duration::from_secs(600);
+// the maximum write kv throughput(20MiB), this is an empirical value.
+const MAX_WRITE_KV_SPEED: u64 = 20 * 1024 * 1024;
+// The maximum duration in seconds we expect IME to release enough memory after
+// memory usage reaches `evict_threshold`. This is an empiracal value.
```
Comment (overvenus) [inline components/in_memory_engine/src/config.rs:21]:
```suggestion
// memory usage reaches `evict_threshold`. This is an empirical value.
```

**Review by overvenus — APPROVED** [review]
Rest LGTM

**Comment by glorv** [comment]
@zhangjinpeng87 PTAL

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**